### PR TITLE
#6Task. valery.aleshka

### PR DESCRIPTION
**Pull request by Valeryaleshka**
https://travis-ci.org/github/Valeryaleshka/js-assignments/builds/715677868

### CODE REVIEW

File: task/06-conditions-n-loops-tasks.js [hunk 1]
```diff
@@ -30,7 +30,19 @@
  *
  */
 function getFizzBuzz(num) {
-    throw new Error('Not implemented');
+
+    if(num%3===0&&num%5===0){
```
Comment (KostyaKov) [inline task/06-conditions-n-loops-tasks.js]: It is necessary to observe indents between operators

File: task/06-conditions-n-loops-tasks.js [hunk 2]
```diff
@@ -30,7 +30,19 @@
...
+    }
+    if(num%3===0){
+        return "Fizz"
+    }
+    if(num%5===0){
+        return "Buzz"
+    }
+    return num;
+
+    //throw new Error('Not implemented');
```
Comment (KostyaKov) [inline task/06-conditions-n-loops-tasks.js]: Unnecessary comments should be removed

File: task/06-conditions-n-loops-tasks.js [hunk 3]
```diff
@@ -63,7 +82,15 @@ function getFactorial(n) {
  *   -1,1  =>  0  ( = -1 + 0 + 1 )
  */
 function getSumBetweenNumbers(n1, n2) {
-    throw new Error('Not implemented');
+
+    let x = 0;
+    for (let i = n1; i <= n2; i++) {
+        x = x + i;
+
```
Comment (KostyaKov) [inline task/06-conditions-n-loops-tasks.js]: Unnecessary spaces should be removed

File: task/06-conditions-n-loops-tasks.js [hunk 4]
```diff
@@ -119,7 +152,22 @@ function isTriangle(a,b,c) {
...
+    let rectangle1 = { leftTop: [rect1.top, rect1.left], rightDown: [rect1.top + rect1.width, rect1.left + rect1.height] }
+    let rectangle2 = { leftTop: [rect2.top, rect2.left], rightDown: [rect2.top + rect2.width, rect2.left + rect2.height] }
+
+    let maxStartWidth = Math.max(rectangle1.leftTop[0], rectangle2.leftTop[0]);
+    let minEndWidth = Math.min(rectangle1.rightDown[0], rectangle2.rightDown[0]);
+    let maxStartHeight = Math.max(rectangle1.leftTop[1], rectangle2.leftTop[1]);
+    let minEndHeight = Math.min(rectangle1.rightDown[1], rectangle2.rightDown[1]);
+
+    if (minEndWidth - maxStartWidth <= 0 || minEndHeight - maxStartHeight <= 0) {
```
Comment (KostyaKov) [inline task/06-conditions-n-loops-tasks.js]: You can return expression result without if statement

File: task/06-conditions-n-loops-tasks.js [hunk 5]
```diff
@@ -192,7 +256,27 @@ function findFirstSingleChar(str) {
...
-    throw new Error('Not implemented');
+
+    let z = Math.min(a,b);
+    let w = Math.max(a,b);
+    let x = "";
+    let y = "";
+
+    if(isStartIncluded){
+        x = x.concat("["+z+", ");
+    }else{
```
Comment (KostyaKov) [inline task/06-conditions-n-loops-tasks.js]: Indentation with curly braces must be respected

File: task/06-conditions-n-loops-tasks.js [hunk 6]
```diff
@@ -150,7 +198,14 @@ function doRectanglesOverlap(rect1, rect2) {
  *   
  */
 function isInsideCircle(circle, point) {
-    throw new Error('Not implemented');
+
+    if (Math.pow(point.x - circle.center.x, 2) + Math.pow(point.y - circle.center.y, 2) < Math.pow(circle.radius, 2)) {
```
Comment (KostyaKov) [inline task/06-conditions-n-loops-tasks.js]: You can return expression result without if statement

File: task/06-conditions-n-loops-tasks.js [hunk 7]
```diff
@@ -270,7 +383,25 @@ function isCreditCardNumber(ccn) {
...
+    }
+    while (z.length > 1) {
+
+        let y = 0;
+        for (let i = 0; i < z.length; i++) {
+            y += parseInt(z[i]);
+        }
+        z = y.toString();
+
+
```
Comment (KostyaKov) [inline task/06-conditions-n-loops-tasks.js]: Extra empty lines can be deleted

File: task/06-conditions-n-loops-tasks.js [hunk 8]
```diff
@@ -332,7 +482,66 @@ function isBracketsBalanced(str) {
...
+    let mins = seconds * 60;
+    let hours = mins * 60;
+    let days = hours * 24;
+    let month = days * 30;
+    let year = month * 12;
+
+    let result = "";
+    let x = endDate - startDate;
+
+    switch (true) {
```
Comment (KostyaKov) [inline task/06-conditions-n-loops-tasks.js]: It is more correct to use "if" conditions

File: task/06-conditions-n-loops-tasks.js [hunk 9]
```diff
@@ -373,7 +585,18 @@ function toNaryString(num, n) {
  *   ['/web/favicon.ico', '/web-scripts/dump', '/webalizer/logs'] => '/'
  */
 function getCommonDirectoryPath(pathes) {
-    throw new Error('Not implemented');
+
+    let A = pathes.concat().sort();
```
Comment (KostyaKov) [inline task/06-conditions-n-loops-tasks.js]: Variables must have lower case

**Review by KostyaKov — COMMENTED** [review]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by Valeryaleshka** [comment]
I've fixed everything.
New Travis-ci build.
https://travis-ci.org/github/Valeryaleshka/js-assignments/builds/721831073

File: task/06-conditions-n-loops-tasks.js [hunk 10]
```diff
@@ -30,7 +30,17 @@
  *
  */
 function getFizzBuzz(num) {
-    throw new Error('Not implemented');
+
+    if(num % 3 === 0 && num % 5 === 0){
```
Comment (KostyaKov) [inline task/06-conditions-n-loops-tasks.js:34]: it is necessary to observe indents after operators
Example: "if (a > b)"

**Review by KostyaKov — COMMENTED** [review]
Closing the PR as accepted! Thanks!

**Pull request closed without merging**